### ISSUE TITLE
What is the purposes of hint?

### ISSUE THREAD

**Issue by tailhook** [kkawakam/rustyline#397]
I'm using hint to hints of the next arguments of the command, like this:
```
some-flag [on|off]
         ^ cursor is here
```
Or even for descriptions of the commands:
```
set -- set configuration value
   ^ cursor is here
```
But in this case `Right` arrow at the end of input puts the hint into the input line. Which is not intended. `Tab` does completion as intended
(And as far as I understand I can't override right arrow to `Complete` because the right arrow in the middle of the input will not work as cursor movement).

Is it wrong usage of hint or is it a bug?

**Comment by gwenn** [comment]
Maybe we should restrict hint to be displayed only when cursor is at the end of input ?

**Comment by tailhook** [comment]
> Maybe we should restrict hint to be displayed only when cursor is at the end of input ?

In my use-cases I do exactly that and still hit this issue.

Also I've plans to hint different arguments, like this:
```
> set-size 100 200  -- *width* height
             ^ cursor is here
> set-size 100 200  -- width *height*
                 ^ cursor is here
```
(where asterisks mark bold or brighter color)
... so restriction isn't what I'm looking for.

**Comment by gwenn** [comment]
Or make the distinction like [completion candidate](https://docs.rs/rustyline/6.1.2/rustyline/completion/trait.Candidate.html) between how the hint is displayed and its replacement when user presses `Right` arrow ?

**Comment by tailhook** [comment]
> Or make the distinction like [completion candidate](https://docs.rs/rustyline/6.1.2/rustyline/completion/trait.Candidate.html) between how the hint is displayed and its replacement when user presses `Right` arrow ?

Yes this sounds good. As this is what I usually do for `Tab` too: sync hint text with completion that is going to happen if user presses Tab key.

**Comment by gwenn** [comment]
Version 7.0.0 released
